### PR TITLE
*: support cluster pause

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

**Comment by xiang90** [comment]
@hongchaodeng 

> FAIL: TestDisasterRecovery2Members

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -151,6 +151,55 @@ func TestDisasterRecoveryAll(t *testing.T) {
...
+	}
+
+	testEtcd.Spec.Paused = true
+	if testEtcd, err = updateEtcdCluster(f, testEtcd); err != nil {
+		t.Fatalf("failed to pause control: %v", err)
+	}
+
+	// TODO: this is used to wait for the TPR to be updated.
+	// TODO: make this wait for reliable
+	time.Sleep(5 * time.Second)
```
Comment (hongchaodeng) [inline test/e2e/e2e_test.go]: Why we wait here? As long as API request returns, it should have been updated?

**Comment by hongchaodeng** [comment]
Failure is related to https://github.com/coreos/etcd-operator/issues/349

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (xiang90) [inline test/e2e/e2e_test.go]: The full TPR update cycle has two phase. API server updates TPR, API server sends TPR updates to the watcher in our operator, the operator sends the update to the cluster monitor routine, the monitor routine finally update its TPR cache (the spec). We need to wait for the full cycle. Basically we can remove this once our operator writes status back. Now we do not.

**Comment by hongchaodeng** [comment]
LGTM
Let me merge this. I will carry the burden to fix test flakes.

**Pull request merged**